collectors/jiraffe/collectors.py

```diff
@@ -0,0 +1,91 @@
+"""
+Jira collector
+"""
+from typing import List, Union
+
+from celery.utils.log import get_task_logger
+from django.utils import timezone
+from jira import Issue
+
+from collectors.framework.models import Collector
+
+from .convertors import TrackerIssueConvertor
+from .core import JiraQuerier
+
+logger = get_task_logger(__name__)
+
+
+class JiraTrackerCollector(Collector, JiraQuerier):
+    """
+    Jira tracker collector
+    """
+
+    # date before the first Jira tracker was created (or at least the oldest update)
+    BEGINNING = timezone.datetime(2014, 1, 1, tzinfo=timezone.get_current_timezone())
+    # Jira API does not seem to have either issues returning large number of results
+    # or any restrictions on the maximum query period so we can be quite greedy
+    BATCH_PERIOD_DAYS = 365
+
+    def get_batch(self) -> (List[Issue], timezone.datetime):
+        """
+        get next batch of Jira trackers plus period_end timestamp
+        """
+        period_start = self.metadata.updated_until_dt or self.BEGINNING
+        period_end = period_start + timezone.timedelta(days=self.BATCH_PERIOD_DAYS)
+        # query for trackers in the period and return them together with the timestamp
+        return self.get_tracker_period(period_start, period_end), period_end
+
+    def collect(self, tracker_id: Union[str, None] = None) -> str:
+        """
+        collector run handler
+
+        on every run the next batch of Jira trackers is fetched and store
+        while proceeding forward by the timestamp of the last update
+
+        tracker_id param makes the collector to sync a tracker of the given ID only
+        """
+
+        logger.info("Fetching Jira trackers")
+        start_dt = timezone.now()
+        desync = []
```

Comment:
should probably be called `updated_trackers` or similar instead

Reply:
I will rename.